core/services/aws/notification/sendPushNotification.ts

```diff
@@ -50,20 +57,63 @@ async function processSQSRecord(record: SQSRecord): Promise<void> {
       new DynamoDbTableOperations<UserDetailsDTO, UserFields, UserModel>(new UserModel())
     )
     userDeviceToSnsEndpointMap.set(userId, userSnsEndpointArn)
+    await createNotification(body)
     await notificationService.publishNotification(
       body,
       userSnsEndpointArn,
-      new DynamoDbTableOperations<NotificationDTO, NotificationFields, NotificationModel>(new NotificationModel()),
       new SNSOperations()
     )
   } else {
+    await createNotification(body)
     await notificationService.publishNotification(
       body,
       cachedUserSnsEndpointArn,
-      new DynamoDbTableOperations<NotificationDTO, NotificationFields, NotificationModel>(new NotificationModel()),
       new SNSOperations()
     )
   }
 }
 
 export default sendPushNotification
+
+async function createNotification(body: NotificationAttributes): Promise<void> {
+  if (['BLOOD_REQ_POST', 'REQ_ACCEPTED'].includes(body.type)) {
+    const notificationData: BloodDonationNotificationAttributes = {
+      type: body.type,
+      payload: {
+        seekerId: body.payload.seekerId as string,
+        requestPostId: body.payload.requestPostId as string,
+        createdAt: body.payload.createdAt as string,
+        bloodQuantity: body.payload.bloodQuantity as string,
+        requestedBloodGroup: body.payload.requestedBloodGroup as string,
+        urgencyLevel: body.payload.urgencyLevel as string,
+        contactNumber: body.payload.contactNumber as string,
+        donationDateTime: body.payload.donationDateTime as string,
+        seekerName: body.payload.seekerName as string,
+        patientName: body.payload.patientName as string,
+        location: body.payload.location as string,
```

Comment:
Why do we need `as` in every property?

Reply:
cause, here notification can be multiple types of data. so when its a blood donation notification I had to handle it in a different way.

Reply:
Please let me know if there is a better way.